openmdao/core/group.py

```diff
@@ -2631,14 +2631,9 @@ def _guess_nonlinear(self):
         complex_step = self._inputs._under_complex_step
 
         if complex_step:
-            self._inputs.set_complex_step_mode(False, keep_real=True)
-            self._residuals.set_complex_step_mode(False, keep_real=True)
-
-            # The Group outputs vector contains imaginary numbers from other components, so we need
-            # to save a cache and restore it later.
-            imag_cache = np.empty(len(self._outputs._data))
-            imag_cache[:] = self._outputs._data.imag
```

Comment:
And the other things we don't need any more are 2 dicts (_views, _views_flat) for every Vector in the entire model, so that's significant as well.  It's not ideal that I had to cache the old values in compute_approximations, and it may be that we could be smarter about that and maybe only do it in certain circumstances, but I tried turning it off after everything was passing and it caused a few failures.